### ISSUE TITLE
Coverage ...

### ISSUE THREAD

**Issue by nelsonic** [dwyl/phoenix-ecto-append-only-log-example#10]
https://codecov.io/github/dwyl/phoenix-ecto-append-only-log-example?branch=master
![image](https://user-images.githubusercontent.com/194400/47010334-969b4f00-d136-11e8-8eab-53832141d95f.png)

![image](https://user-images.githubusercontent.com/194400/47010349-a31fa780-d136-11e8-990e-8d32865f5913.png)

# Todo

+ [ ] do better.

**Comment by nelsonic** [comment]
P.S: I _never_ ask anyone to do something I'm not prepared to do _myself_: 
https://codecov.io/github/dwyl/phoenix-chat-example?branch=master
![image](https://user-images.githubusercontent.com/194400/47010449-da8e5400-d136-11e8-8399-a69b99142384.png)

**Comment by nelsonic** [comment]
on `localhost`, the result of running the command `MIX_ENV=test mix cover` is:
![image](https://user-images.githubusercontent.com/194400/47040969-8360a180-d17f-11e8-9fec-d7e43179dc53.png)

Clearly we have some work to do to write a few more test ... 👷

**Comment by Danwhy** [comment]
We shouldn't need to write any more tests, the only lines that aren't covered are autogenerated Phoenix code

**Comment by nelsonic** [comment]
@Danwhy yeah, I'm _mostly_ seeing that ... _however_ excoveralls thinks the Macros aren't being used:
![image](https://user-images.githubusercontent.com/194400/47076276-44216780-d1f6-11e8-9195-f6447fd1f9de.png)

**Comment by Danwhy** [comment]
Yeah unfortunately a known issue: https://github.com/parroty/excoveralls/issues/59

**Comment by nelsonic** [comment]
@Danwhy yeah, sucks that Macros don't get "instrumented" for coverage. 
Does it mean we need to switch to using "normal" Modules/Functions? see: #8

**Comment by Danwhy** [comment]
It is something to consider. I'll write up some of the pros and cons of each approach when answering #8
